docs/book/src/reference/project-config.md

```diff
@@ -2,7 +2,74 @@
+    resources:
+    - domain: testproject.org
+      group: example.com
+      kind: Memcached
+      options:
+        containerCommand: memcached,-m=64,-o,modern,-v
+        containerPort: "11211"
+        image: memcached:1.4.36-alpine
+        runAsUser: "1001"
+      version: v1alpha1
+    - domain: testproject.org
+      group: example.com
+      kind: Busybox
+      options:
+        image: busybox:1.28
+      version: v1alpha1
+projectName: project-v4-with-deploy-image
+repo: sigs.k8s.io/kubebuilder/testdata/project-v4-with-deploy-image
+resources:
+- api:
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: testproject.org
+  group: example.com
+  kind: Memcached
+  path: sigs.k8s.io/kubebuilder/testdata/project-v4-with-deploy-image/api/v1alpha1
+  version: v1alpha1
+  webhooks:
+    validation: true
+    webhookVersion: v1
+- api:
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: testproject.org
+  group: example.com
+  kind: Busybox
+  path: sigs.k8s.io/kubebuilder/testdata/project-v4-with-deploy-image/api/v1alpha1
+  version: v1alpha1
+version: "3"
+```
+## Why do we need to store the plugins and data used?
+
+Kubebuilder allows users to scaffold the project with external plugins and the PROJECT file can track the information required for the same, for example: 
+- we can check if a plugin can or cannot be scaffolded on top of an existing plugin (ie) plugin compatibility while chaining multiple of them together.
+- what operations can or cannot be done for ex. can multiple APIs be scaffolded for the current configuration or not.
+- what data can or not be used in the CLI operations for ex. does an API exist in the project before scaffolding webhooks for the same.
+
+Note that KubeBuilder is not only a CLI tool but can also be used as a library to allow users to create their plugins/tools, provide helpers and customizations on top of their existing projects - an example of which is [Operator-SDK](https://sdk.operatorframework.io/). SDK leverages KubeBuilder to create plugins to allow users to work with other languages and provide helpers for their users to integrate their projects with the [Operator Framework solutions/OLM](https://olm.operatorframework.io/). You can check the [plugin's documentation](https://book.kubebuilder.io/plugins/creating-plugins.html#why-use-the-kubebuilder-style) to know more about creating custom plugins.
+
+Additionally, another motivation for the PROJECT file is to help us to create a feature that allows users to easily upgrade their projects by providing helpers that automatically re-scaffold the project. By having all the required metadata regarding the APIs, their configurations and versions in the PROJECT file, we could automate the process of re-scaffolding while migrating between plugin versions. More info on this, can be found [here] (https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/helper_to_upgrade_projects_by_rescaffolding.md)
```

Comment:
```suggestion
Additionally, another motivation for the PROJECT file is to help us to create a feature that allows users to easily upgrade their projects by providing helpers that automatically re-scaffold the project. By having all the required metadata regarding the APIs, their configurations and versions in the PROJECT file. For example, it can be used to automate the process of re-scaffolding while migrating between plugin versions. ([More info][doc-design-helper]).
```